From f89e76119ecd96d1f4787c66e298c4a2c76a985f Mon Sep 17 00:00:00 2001
From: Artem Senichev <a.senichev@yadro.com>
Date: Sat, 11 Jul 2020 11:30:21 +0300
Subject: [PATCH] Handle big network latency during DHCP requests

Petitboot loads the PXE configuration file after an IP address is binded
to the network interface, but before it will be assigned. In a network
with big latencies, this leads to using the default (link-local) address
during the loading procedure.
Unfortunately, the current implementation of udhcpc6 does not support
the "Assignment" event, and as a result, Petitboot can not work properly
in such networks.

This patch adds a 3-second pause between binding the IP address and
loading the configuration file.

Signed-off-by: Artem Senichev <a.senichev@yadro.com>
---
 discover/pxe-parser.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/discover/pxe-parser.c b/discover/pxe-parser.c
index 035794c..6c42394 100644
--- a/discover/pxe-parser.c
+++ b/discover/pxe-parser.c
@@ -515,6 +515,7 @@ static int pxe_parse(struct discover_context *dc)
 			pb_url_to_string(conf->dc->conf_url));
 
 		/* we have a complete URL; use this and we're done. */
+		sleep(3);
 		result = load_url_async(conf->dc, file_url,
 					pxe_conf_parse_cb, conf, NULL, ctx);
 		if (!result) {
-- 
2.27.0

